pyfstat/core.py

```diff
@@ -955,44 +982,50 @@ def calculate_twoF_cumulative(
         argp=None,
         tstart=None,
         tend=None,
-        npoints=1000,
+        num_segments=1000,
     ):
-        """Calculate the cumulative twoF along the obseration span
+        """Calculate the cumulative twoF over subsets of the observation span.
+
+        This means that we consider sub-"segments" of the [tstart,tend] interval,
+        each starting at the overall tstart and with increasing durations,
+        and compute the 2F for each of these, which for a true CW signal should
+        increase roughly with duration towards the full value.
 
         Parameters
         ----------
         F0, F1, F2, Alpha, Delta: float
             Parameters at which to compute the cumulative twoF
         asini, period, ecc, tp, argp: float, optional
-            Binary parameters at which to compute the cumulative 2F
-        tstart, tend: int
-            GPS times to restrict the range of data used - automatically
-            truncated to the span of data available
-        npoints: int
-            Number of points to compute twoF along the span
-
-        Notes
-        -----
-        The minimum cumulatibe twoF is hard-coded to be computed over
-        the first 6 hours from either the first timestampe in the data (if
-        tstart is smaller than it) or tstart.
+            Optional: Binary parameters at which to compute the cumulative 2F
+        tstart, tend: int or None
+            GPS times to restrict the range of data used;
+            if None: falls back to self.minStartTime and self.maxStartTime;
+            if outside those: auto-truncated
+        num_segments: int
+            Number of segments to split [tstart,tend] into
+
+        Returns
+        -------
+        cumulative_durations : ndarray of shape (num_segments,)
+            Offsets of each segment's tend from the overall tstart.
+        twoFs : ndarray of shape (num_segments,)
+            Values of twoF computed over
+            [[tstart,tstart+duration] for duration in cumulative_durations].
 
         """
-        SFTminStartTime = self.SFT_timestamps[0]
-        SFTmaxStartTime = self.SFT_timestamps[-1]
-        tstart = np.max([SFTminStartTime, tstart])
-        min_tau = np.max([SFTminStartTime - tstart, 0]) + 3600 * 6
-        max_tau = SFTmaxStartTime - tstart
-        taus = np.linspace(min_tau, max_tau, npoints)
-        twoFs = []
         if not self.transientWindowType:
-            # still call the transient-Fstat-map function, but using the full range
-            self.transientWindowType = "none"
+            # FIXME: Can we move this elsewhere?
```

Comment:
why?

Reply:
This method is re-initializing the whole instance with a different window.
Should we assume a "Hey user, make sure not to re-use your instances: always create a new one from scratch" way of thinking? That's fine with me, but at least we should ad a logging or something acknowledging the window change + initialization.

Reply:
It's ugly and a bit dangerous, yes. I think we use similar hacks in several places, but that's no real excuse.

I think the easiest solution is to _reset it at the end of this function_ (store current `self.transientWindowType` in a temporary variable to do so).

Reply:
That defeats the purpose... since the reason to do it here is to make sure one can use the transient trick to compute "cumulative 2F" as "Transient 2F with increasing lenght"?

Reply:
cd4b13c will do the trick until we clean up this class structure (and by "clean up" I mean "tear down and re-build re-using as much code as possible"; but that's more of a dream than a reality rn).